### ISSUE TITLE
Host Haiku translations

### ISSUE THREAD

**Issue by bitigchi** [KapiX/Polyglot#76]
On IRC, during a rant/brainstorming session about the fate of Polyglot and other translation problems, @waddlesplash had this idea, why don't we use Polyglot to host Haiku core translations? Surely it would require some plumbing and front-faceting, but it would really be cool!

Interface requirements are a different thing however, a big UI revamp is needed...

**Comment by KapiX** [comment]
I agree it would be cool. As you mentioned, UI needs to be changed first. Mockups welcome :)

I would see it as an additional type of view (configured per project), since the existing one works well for apps.

Concerns:
- performance - I did not perform stress tests
- contributor trail - Polyglot only stores contributors for projects in a way that doesn't allow to assign each contributor to the translation they contributed. Only last translation version stores who contributed it. However, even if there are no current translations by someone, if they ever added/modified one, they will still be listed. This was okay so far, but maybe for Haiku there are different requirements. Those need to be discussed. (Some systems use Git, but I'm not sure this is the best option.)
- perhaps some kind of info whether someone is currently editing the translation will have to be implemented.

Work to do before migration starts:
- APIs for mass uploads. (AFAIK Pootle gets one ZIP file with all texts.)
- Job queues - files need to be processed asynchronously when there are a lot of them uploaded at 
- Suggestions

**Comment by bitigchi** [comment]
Some UI ideas:

- Visual new lines
- Source and target boxes on top of each other
- Monospaced font
- 80-column mark (useful for console help text)
- Tab and space visualisers
- Easy-insert character set
  - Dashes (en, em, figure, minus sign)
  - Ellipsis (two, three dots)
  - Typographic quotes (bonus if it can display variants according to the language)
  - Space variants (no break space, narrow no break space, zero width space, zero width joiner)
- When there are multiple files in a project, the UI is not clear, asks them to download, and it’s always needed to "show all translations”
- Other languages other than the translator’s preferred language can be hidden in a sidebar or another page
- Easy jump to next string, keyboard shortcut marks for each button for easy grasp
